### PR TITLE
fix: deploy jvm-build-service custom images

### PR DESCRIPTION
**Pull request by psturc**
### JIRA
https://issues.redhat.com/browse/PLNSRVCE-836

### How was this tested?
```bash
export JVM_BUILD_SERVICE_IMAGE_REPO=quay.io/redhat-appstudio/hacbs-jvm-controller \
JVM_BUILD_SERVICE_IMAGE_TAG=f1ccc0358907c305dc0fe9b5105f17c9a0592755 \
JVM_BUILD_SERVICE_PR_OWNER=psturc \
JVM_BUILD_SERVICE_PR_SHA=f1ccc0358907c305dc0fe9b5105f17c9a0592755 \
JVM_BUILD_SERVICE_CACHE_IMAGE=quay.io/redhat-appstudio/hacbs-jvm-cache:f1ccc0358907c305dc0fe9b5105f17c9a0592755 \
JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE=quay.io/redhat-appstudio/hacbs-jvm-build-request-processor:f1ccc0358907c305dc0fe9b5105f17c9a0592755
./hack/bootstrap-cluster.sh preview
```
[Result](https://github.com/psturc/infra-deployments/commit/9561b01673767f3bf70de6bb9531560b9f7b72c3#diff-bf4f7ec522f073ba099ecf8bc6b927ebe7ff9fc3aabc11e788bb97708bf1dfe8)

### CODE REVIEW

File: hack/preview-template.env [hunk 1]
```diff
@@ -30,19 +30,8 @@ export JVM_BUILD_SERVICE_IMAGE_REPO=
...
-export JVM_BUILD_SERVICE_SIDECAR_IMAGE_TAG=
-export JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE_REPO=
-export JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE_TAG=
-export JVM_DELETE_TASKRUN_PODS=
-export JVM_BUILD_SERVICE_JDK8_BUILDER_IMAGE_REPO=
-export JVM_BUILD_SERVICE_JDK8_BUILDER_IMAGE_TAG=
-export JVM_BUILD_SERVICE_JDK11_BUILDER_IMAGE_REPO=
-export JVM_BUILD_SERVICE_JDK11_BUILDER_IMAGE_TAG=
-export JVM_BUILD_SERVICE_JDK17_BUILDER_IMAGE_REPO=
-export JVM_BUILD_SERVICE_JDK17_BUILDER_IMAGE_TAG=
```
Comment (psturc) [inline hack/preview-template.env:45]: I checked jvm-build-service and didn't find any reference to these env vars, so I assumed they are not used anymore. cc @stuartwdouglas for double check please

File: hack/preview.sh [hunk 2]
```diff
@@ -105,14 +105,7 @@ yq -i e ".data |= .\"transparency.url\"=\"https://$rekor_server\"" $ROOT/compone
...
-[[ -n "${JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE_REPO}" && ${JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE_TAG} ]] && yq -i e "(.spec.template.spec.containers[].env[] | select(.name==\"JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE\")) |=.value=\"${JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE_REPO}:${JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE_TAG}\"" $ROOT/components/jvm-build-service/operator-images.yaml
+[[ -n "${JVM_BUILD_SERVICE_CACHE_IMAGE}" && ${JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE} ]] && patch=$(yq "(.patches[] | select(.target.kind == \"Deployment\")).patch" $ROOT/components/jvm-build-service/kustomization.yaml | yq e "select(.[].path == \"/spec/template/spec/containers/0/env\") | .[].value |= . + [{\"name\" : \"JVM_BUILD_SERVICE_CACHE_IMAGE\", \"value\": \"${JVM_BUILD_SERVICE_CACHE_IMAGE}\"}, {\"name\": \"JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE\", \"value\": \"${JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE}\"}] | (.[].value[] | select(.name == \"IMAGE_TAG\")) |= .value = \"\"") yq -i e "(.patches[] | select(.target.kind == \"Deployment\")).patch |= strenv(patch)" $ROOT/components/jvm-build-service/kustomization.yaml
```
Comment (psturc) [inline hack/preview.sh:108]: this might be a bit hardcode one-liner. any suggestions for improvement are welcome 😅

Comment (psturc) [reply hack/preview.sh:108]: ok then we'd have to update the [script for updating jvm-build-service image tag](https://github.com/redhat-appstudio/jvm-build-service/blob/f1ccc0358907c305dc0fe9b5105f17c9a0592755/.tekton/controller-push.yaml#L23), right?

Comment (psturc) [reply hack/preview.sh]: updated, [result](https://github.com/psturc/infra-deployments/commit/668d3d2456a862362dce5da65121e7f2a3ae02da#diff-44030e8cb4fc8790e75c962d72e7a46457f9c4ed86e6fd554d206db955dc5685)

**Comment by psturc** [comment]
@Michkov @stuartwdouglas ok to merge?

**Review by Michkov — APPROVED** [review]

**Pull request merged**